lib/services/deviceService.js

```diff
@@ -153,11 +157,85 @@ function sendRegistrations(unregister, deviceData, callback) {
 function processContextRegistration(deviceData, body, callback) {
     var newDevice = _.clone(deviceData);
 
-    newDevice.registrationId = body.registrationId;
+    if (body) {
+        newDevice.registrationId = body.registrationId;
+    }
 
     callback(null, newDevice);
 }
 
+
+function createInitialEntityHandler(deviceData, newDevice, callback) {
+    return function handleInitialEntityResponse(error, response, body) {
+        if (error) {
+            logger.error(context, 'Connection error creating inital entity in the Context Broker: %s', error);
+            callback(error);
+        } else if (response && body && response.statusCode === 200) {
+            var errorField = ngsiParser.getErrorField(body);
+
+            if (errorField) {
+                logger.error(context, 'Update error connecting to the Context Broker: %j', errorField);
+                callback(new errors.BadRequest(JSON.stringify(errorField)));
+            } else {
+                logger.debug(context, 'Initial entity created successfully.');
+                callback(null, newDevice);
+            }
+        } else {
+            var errorObj;
+
+            logger.error(context, 'Protocol error connecting to the Context Broker: %j', errorObj);
+
+            errorObj = new errors.EntityGenericError(deviceData.id, deviceData.type, errorObj);
+
+            callback(errorObj);
+        }
+    };
+}
+
+function createInitialEntity(deviceData, newDevice, callback) {
+    var options = {
+        url: 'http://' + config.contextBroker.host + ':' + config.contextBroker.port + '/v1/updateContext',
```

Comment:
Maybe the path could also be configurable... Mainly since an API change in the Context Broker is coming... :O Once again, I leave it to your discretion since it is not really that important... ;)